develop/app/Anser/Services/v2/ProductService.php

```diff
@@ -16,9 +16,64 @@ class ProductService extends SimpleService
     protected $retryDelay = 1;
     protected $timeout    = 5.0;
 
-    public function getAllProduct()
+    /**
+     * Get all product
+     *
+     * @param integer|null $limit
+     * @param integer|null $offset
+     * @param string|null $isDesc
+     * @return ActionInterface
+     */
+    public function getAllProduct(?int $limit, ?int $offset, ?string $isDesc): ActionInterface
```

Comment:
If there are the `null` type of parameters, I strongly recommend you should defind these parameters the initial values for them. shown below:
> public function getAllProduct(?int $limit = null, ?int $offset = null, ?string $isDesc = null): ActionInterface

Then the user can can this method without any parameters, like: `getAllProduct()`.

Reply:
This change is done.